packages/itwinui-react/src/core/SideNavigation/SideNavigation.tsx

```diff
@@ -113,6 +113,7 @@ export const SideNavigation = React.forwardRef((props, forwardedRef) => {
 
   const ExpandButton = (
     <IconButton
+      label='Expand or collapse side menu'
```

Comment:
i'd go with "Toggle icon labels". It's more accurate and concise.

also `aria-expanded` for assistive tech

```suggestion
      label='Toggle icon labels'
      aira-expanded={_isExpanded}
```